src/js/App/GlobalFilter/GlobalFilterMenu.js

```diff
@@ -0,0 +1,164 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { TextInput, MenuList, MenuItem, Select, SelectVariant, MenuGroup, Checkbox } from '@patternfly/react-core';
+
+import './global-filter-menu.scss';
+
+export const groupType = {
+  checkbox: 'checkbox',
+  radio: 'radio',
+  button: 'button',
+  plain: 'plain',
+};
+
+const getMenuItems = (groups, filterValueRegex, onChange, calculateSelected) => {
+  const result = groups.map(({ value, label, id, type, items, ...group }) => ({
+    label,
+    value,
+    items: items.map((item, index) => ({
+      ...item,
+      key: item.id || item.value || index,
+      value: String(item.value || item.id || index),
+      onClick: (event) => {
+        onChange(
+          event,
+          calculateSelected(type, value, item.value),
+          {
+            value,
+            label,
+            id,
+            type,
+            items,
+            ...group,
+          },
+          item,
+          value,
+          item.value
+        );
+      },
+    })),
+  }));
+  return result.filter(({ noFilter, items = [] }) => noFilter || items.length > 0);
+};
+
+const GlobalFilterMenu = (props) => {
+  const { filterBy, onFilter, groups = [], onChange, selectedTags } = props;
+  const [isOpen, setIsOpen] = useState(false);
+
+  const calculateSelected = useCallback(
+    (type, groupKey, itemKey) => {
+      const activeGroup = selectedTags[groupKey];
+      if (activeGroup) {
+        if (type !== groupType.radio && (activeGroup[itemKey] instanceof Object ? activeGroup[itemKey].isSelected : Boolean(activeGroup[itemKey]))) {
+          return {
+            ...selectedTags,
+            [groupKey]: {
+              ...(activeGroup || {}),
+              [itemKey]: false,
+            },
+          };
+        }
+
+        return {
+          ...selectedTags,
+          [groupKey]: {
+            ...(type !== groupType.radio ? activeGroup || {} : {}),
+            [itemKey]: true,
+          },
+        };
+      }
+
+      return {
+        ...selectedTags,
+        [groupKey]: {
+          [itemKey]: true,
+        },
+      };
+    },
+    [selectedTags]
```

Comment:
What if you select multiple tags? Does it recalculate?

Reply:
Yeah, you will get a new object.